examples/prometheus/rules/os.rules

```diff
@@ -0,0 +1,26 @@
+groups:
+- name: os-rules
+  interval: 30s # defaults to global interval
+  rules:
+
+# per etcd alert set https://gist.github.com/brancz/eb25f016a663eebd1f44b0261e577874
+    - record: instance:fd_utilization
+      expr: process_open_fds / process_max_fds
+
+    - alert: FdExhaustionIn4Hrs
```

Comment:
predict_linear() alerts won't fire when the process has exhausted (or is about to exhaust) all file descriptors. You would need an alert like "instance:fd_utilization >= 0.99" for those cases.

Reply:
> predict_linear() alerts won't fire when the process has exhausted (or is about to exhaust) all file descriptors.

I'm not sure I understand. Admittedly I'm lacking the benefit from much real-world data since we don't have node exporter deployed broadly yet. I **could** just pull this os.yaml file out for now as we get more hands-on experience.

Reply:
Ok, that makes sense. A picture is worth a thousand words! Thanks for the description.

Reply:
@simonpasquier would you agree predict_linear() is useful for canary-type warning ("You're headed for danger") but critical alerting on resource saturation should use the bare utilization value?

Reply:
@aweiteka that's correct. predict_linear() works fine when the metric's trend is more or less steady. As sometimes it's more erratic, you still need a safety belt with a fixed threshold close to the max saturation level.